engine/language-server/src/main/scala/org/enso/languageserver/search/handler/InvalidateModulesIndexHandler.scala

```diff
@@ -69,22 +77,22 @@ object InvalidateModulesIndexHandler {
   /** Creates a configuration object used to create [[InvalidateModulesIndexHandler]].
     *
     * @param runtimeFailureMapper mapper for runtime failures
-    * @param timeout request timeout
     * @param runtime reference to the runtime connector
     * @param suggestionsHandler reference to the suggestions handler
+    * @param timeout soft request timeout for detecting abnormal behavior
```

Comment:
It such an abnormal behavior happens, it would be good to start [self sampling](https://github.com/enso-org/enso/blob/07793f576b40043f17bba85dd93dd2e5e7296ffb/lib/scala/profiling-utils/src/main/java/org/enso/profiling/sampler/OutputStreamSampler.java) to help us understand what went wrong.

Reply:
We know what's wrong. It was waiting on a lock.